### ISSUE TITLE
Error from parameter assignability incorrectly reported on satisfies keyword 

### ISSUE THREAD

**Issue by ssalbdivad** [microsoft/TypeScript#56433]
### 🔎 Search Terms

satisfies, error, reporting, span, call, function

### 🕗 Version & Regression Information

- This changed between versions 5.0 and 5.1

### ⏯ Playground Link

https://tsplay.dev/mMl61N

### 💻 Code

```ts
const f = (s: { a: true }) => {}

// I would expect the error to span:
// a) "{}" (Ideal IMO)
// b) "{} satisfies unknown" (A little messier but not wrong- this was the 5.0 behavior)

// The error is instead reported only on "satisfies" despite {} satisfying unknown
f({} satisfies unknown)
```

### 🙁 Actual behavior

(see example)

### 🙂 Expected behavior

(see example)

**Comment by Andarist** [comment]
I'm pretty certain that this is a fallout from one of my PRs: https://github.com/microsoft/TypeScript/pull/53797

I think that the solution would be to "unwrap" the `errorNode` in argument positions to the `satisfies`' expression. I'll put up a PR for this later, I just need to locate all of the places that have to be updated and add the relevant test cases.
